src/integration/java/io/pinecone/integration/controlPlane/serverless/CreateDescribeListAndDeleteIndexTest.java

```diff
@@ -1,52 +1,131 @@
+        assertNotNull(indexModel.getSpec().getServerless());
 
         // List the index
         IndexList indexList = controlPlaneClient.listIndexes();
         assertNotNull(indexList.getIndexes());
+        assertTrue(indexList.getIndexes().stream().anyMatch(index -> indexName.equals(index.getName())));
+    }
 
-        // Delete the index
-        controlPlaneClient.deleteIndex(indexName);
-        Thread.sleep(3500);
+    @Test
+    public void createIndexWithInvalidName() {
+        ServerlessSpec serverlessSpec = new ServerlessSpec().cloud(ServerlessSpec.CloudEnum.AWS).region("us-west-2");
+        CreateIndexRequestSpec createIndexRequestSpec = new CreateIndexRequestSpec().serverless(serverlessSpec);
+        CreateIndexRequest createIndexRequest = new CreateIndexRequest()
+                .name("Invalid-name")
+                .metric(IndexMetric.COSINE)
+                .dimension(10)
+                .spec(createIndexRequestSpec);
+
+        try {
+            controlPlaneClient.createIndex(createIndexRequest);
+
+            fail("Expected to throw PineconeBadRequestException");
+        } catch (PineconeBadRequestException expected) {
+            assertTrue(expected.getLocalizedMessage().contains("must consist of lower case alphanumeric characters or '-'"));
+        }
+    }
+
+    @Test
+    public void createIndexWithInvalidDimension() {
+        ServerlessSpec serverlessSpec = new ServerlessSpec().cloud(ServerlessSpec.CloudEnum.AWS).region("us-west-2");
+        CreateIndexRequestSpec createIndexRequestSpec = new CreateIndexRequestSpec().serverless(serverlessSpec);
+        CreateIndexRequest createIndexRequest = new CreateIndexRequest()
+                .name("invalid-dimension")
+                .metric(IndexMetric.COSINE)
+                .dimension(-1)
+                .spec(createIndexRequestSpec);
+
+        try {
+            controlPlaneClient.createIndex(createIndexRequest);
+
+            fail("Expected to throw PineconeUnmappedHttpException");
+        } catch (PineconeUnmappedHttpException expected) {
+            assertTrue(expected.getLocalizedMessage().contains("dimension: invalid value"));
+        }
+    }
+
+    @Test
+    public void createIndexInvalidCloud() {
+        ServerlessSpec serverlessSpec = new ServerlessSpec().cloud(ServerlessSpec.CloudEnum.AZURE).region("us-west-2");
+        CreateIndexRequestSpec createIndexRequestSpec = new CreateIndexRequestSpec().serverless(serverlessSpec);
+        CreateIndexRequest createIndexRequest = new CreateIndexRequest()
+                .name("invalid-cloud")
+                .metric(IndexMetric.COSINE)
+                .dimension(1)
+                .spec(createIndexRequestSpec);
+
+        try {
+            controlPlaneClient.createIndex(createIndexRequest);
+
+            fail("Expected to throw PineconeNotFoundException");
+        } catch (PineconeNotFoundException expected) {
+            assertTrue(expected.getLocalizedMessage().contains("not found"));
```

Comment:
Not specific to this test, but maybe we should log a backlog issue to make the error msg for an invalid cloud more helpful than `"not found"`? Maybe something even just mentioning the resource, so like `"cloud provider not found"` and then an outlink to our documentation about current cloud providers? Not sure where to put that feedback...

Reply:
(same w/region)

Reply:
Most of these errors are being returned from the server, so any tickets for adjusting the messages on these would need to be handled from that perspective. We're not generating the vast majority of these strings in the client.